src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -104,7 +104,9 @@ private Assembly TryResolveAssembly(AssemblyLoadContext context, AssemblyName as
                 {
                     foreach (var extension in _extensions)
                     {
-                        var candidatePath = Path.Combine(dependencyPath, $"{assemblyName.Name}.{extension}");
+                        var candidatePath = Path.Combine(dependencyPath,
+                            assemblyName.CultureName ?? string.Empty,
```

Comment:
This is based on the assumption that only satellite assemblies have a culture set on them. If the assumption breaks this might lead to task dependencies not being found, if they have a culture set on them.